### PR TITLE
app-admin/flannel: bump to v0.6.1

### PR DESCRIPTION
**Pull request by mischief**
cleanup ebuild a bit, remove old files and respect target arch in
default container image tag

fixes coreos/bugs#1532

### CODE REVIEW

**Comment by marineam** [comment]
The tc-arch bit looks good, though are we going to switch to using arch-aware ACI fetching at some point?

**Comment by crawford** [comment]
> are we going to switch to using arch-aware ACI fetching at some point?

Isn't `rkt fetch` arch-aware already? Quay.io might not be, but the coreos.com domain will support it.

**Comment by glevand** [comment]
LGTM

**Comment by marineam** [comment]
@crawford it isn't arch aware if it is fetching a docker image, which I think is still the case here?

**Comment by marineam** [comment]
It merely says "quay.io/coreos/flannel" which may be ACI but last I knew quay generated ACIs on the fly from docker images.

**Comment by crawford** [comment]
> quay generated ACIs on the fly from docker images

That's correct. We'd switch the url to "coreos.com/flannel:v0.6.1" and update the metadata on coreos.com to point at GitHub (which does have the published ACIs).

**Comment by mischief** [comment]
currently aci discovery for flannel is busted:

```
$ rkt --debug fetch coreos.com/flannel:v0.6.1
image: searching for app image coreos.com/flannel
fetch: 
  ??discovery failed for "coreos.com/flannel"
    ??discovery failed
```

and since i tested that, i decided to test the docker image too which also appears busted!

```
$ docker pull quay.io/coreos/flannel-amd64:v0.6.1
Pulling repository quay.io/coreos/flannel-amd64
Error: Status 403 trying to pull repository coreos/flannel-amd64: "{\"error\": \"Permission Denied\"}"
```

**Comment by crawford** [comment]
> currently aci discovery for flannel is busted

coreos.com does not have the metadata for flannel. Easy fix.

**Comment by mischief** [comment]
ok, i will switch the repo and drop tc-arch.

**Comment by mischief** [comment]
switched repos and dropped `tc-arch`. waiting for `coreos.com/flannel` aci discovery and @tomdee to publish github.com/coreos/flannel aci signatures.

**Comment by crawford** [comment]
After talking with @philips, we've decided to not adopt ac-discovery for this wrapper. Let's stick with the original implementation.

**Comment by steveeJ** [comment]
> After talking with @philips, we've decided to not adopt ac-discovery for this wrapper. Let's stick with the original implementation.

Would you mind explaining this decision?

**Comment by crawford** [comment]
@steveeJ please follow up with @philips offline.

**Comment by tomdee** [comment]
@mischief please LMK if you still need anything from me on this.

**Comment by mischief** [comment]
@tomdee we're going to go with quay.io so there is no need for signing currently. thanks.

File: app-admin/flannel/files/flanneld.service [hunk 1]
```diff
@@ -1,42 +1,53 @@
...
 
 [Service]
 Type=notify
 Restart=always
 RestartSec=5
 Environment="TMPDIR=/var/tmp/"
-Environment="DOCKER_HOST=unix:///var/run/early-docker.sock"
 Environment="FLANNEL_VER={{flannel_ver}}"
-Environment="FLANNEL_IMG=quay.io/coreos/flannel"
+Environment="FLANNEL_IMG=coreos.com/flannel"
```
Comment (crawford) [inline app-admin/flannel/files/flanneld.service]: Use quay.

**Comment by mischief** [comment]
ptal

**Comment by crawford** [comment]
Can you break this into multiple commits?

File: app-admin/flannel/flannel-9999.ebuild [hunk 2]
```diff
@@ -9,7 +9,7 @@ else
 	KEYWORDS="amd64"
 fi
 
-inherit systemd
+inherit systemd toolchain-funcs
```
Comment (marineam) [inline app-admin/flannel/flannel-9999.ebuild]: toolchain-funcs unused

File: app-admin/flannel/flannel-9999.ebuild [hunk 3]
```diff
@@ -23,7 +23,7 @@ RDEPEND="app-emulation/rkt"
 S="$WORKDIR"
 
 src_install() {
-	sed "s/{{flannel_ver}}/${PV}/" "${FILESDIR}"/flanneld-rkt.service >"${T}"/flanneld.service
+	sed --expression "s/{{flannel_ver}}/v${PV}-${ARCH}/" "${FILESDIR}"/flanneld.service >"${T}"/flanneld.service
```
Comment (marineam) [inline app-admin/flannel/flannel-9999.ebuild]: -e is fine

**Comment by mischief** [comment]
ptal

**Pull request merged**